services/venice-controller/src/main/java/com/linkedin/venice/controller/DeferredVersionSwapService.java

```diff
@@ -870,12 +946,43 @@ private Runnable getRunnableForDeferredVersionSwap() {
             break;
           }
 
+          ThreadPoolExecutor clusterExecutorService = getOrCreateExecutorForCluster(cluster);
+          ThreadPoolStats clusterThreadPoolStats = clusterToThreadPoolStatsMap.get(cluster);
+
+          // Filter out stores that are already being processed to prevent duplicate processing
+          List<Store> eligibleStoresToProcess = new ArrayList<>();
+          for (Store parentStore: parentStores) {
+            String kafkaTopicName =
+                Version.composeKafkaTopic(parentStore.getName(), parentStore.getLargestUsedVersionNumber());
+            if (tryStartProcessingStore(kafkaTopicName) && isTargetRegionPushWithDeferredSwapEnabled(parentStore)) {
```

Comment:
should we move this check `tryStartProcessingStore` in before submit to the threadpool ?